Core/Src/bms_can_monitor.c

```diff
@@ -0,0 +1,51 @@
+#include "bms_can_monitor.h"
+#include "timer.h"
+#include "fault.h"
+#include "can.h"
+#include "cerberus_conf.h"
+#include <assert.h>
+#include <stdlib.h>
+
+#define BMS_WATCHDOG_DURATION 10 /*Duration of btween petting bms monitor watchdog*/
+#define CAN_QUEUE_SIZE 5
+
+osThreadId_t bms_can_monitor_handle;
+const osThreadAttr_t bms_can_monitor_attributes = {
+	.name = "BMSCANMonitor",
+	.stack_size = 128 * 8,
+	.priority = (osPriority_t)osPriorityLow1 /*Adjust priority*/
+};
+
+bms_can_monitor_t* bms_can_monitor_init() {
+
+    bms_can_monitor_t* bms_can_monitor = malloc(sizeof(bms_can_monitor_t));
+	assert(bms_can_monitor);
+
+    bms_can_monitor_queue = osMessageQueueNew(CAN_QUEUE_SIZE, sizeof(can_msg_t), NULL);
+    assert(bms_can_monitor_queue);
+
+    return bms_can_monitor;
+}
```

Comment:
When did this monitor stuff get added? Definitely unnecessary for what can be accomplished with a variable inside the task

Reply:
Like the struct is probably unnecessary

Reply:
Actually I take this back, but we should be making this the initialization of a whole bms struct rather than just the CAN monitor. Same logic just rename variables